### PR TITLE
Fix SliverAppBar title opacity and test all cases

### LINKED ISSUE #25993: Title of pinned, floating SliverAppBar not fading out with "bottom"
## Steps to Reproduce

The "issue"/(unexpected behavior) that can be seen is that the title of the `SliverAppBar` no longer fades out when it's pinned and floating when the "bottom" has a widget.

Previously (current beta/stable) the title slowly faded out when the `NestedScrollView` is scrolled and the bottom "overlaps" the title which was almost invisible by the time the bottom was at the top.

Now the title no longer fades out when the `NestedScrollView` is scrolled and the "bottom" basically overlaps the title with no transition.

(See videos // screenshots)

This is probably related to: #25051 @goderbauer 

Code to reproduce (try on current dev channel and on current beta/stable => Flutter 1.0.0):
```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage();

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: NestedScrollView(
          headerSliverBuilder: (context, innerBoxScrolled) => [
                SliverAppBar(
                  title: Text("Sliver App Bar"),
                  floating: true,
                  pinned: true,
                  bottom: PreferredSize(
                    preferredSize: Size.fromHeight(35.0),
                    child: Padding(
                      padding: EdgeInsets.fromLTRB(8.0, 0.0, 8.0, 8.0),
                      child: Container(
                        width: 144.0,
                        child: TextField(
                          maxLines: 1,
                          textAlign: TextAlign.center,
                          decoration: InputDecoration(
                            hintText: 'Text Field',
                            focusedBorder: UnderlineInputBorder(borderSide: BorderSide(color: Colors.white)),
                            border: UnderlineInputBorder(borderSide: BorderSide(color: Colors.white)),
                            contentPadding: EdgeInsets.all(2.0),
                          ),
                        ),
                      ),
                    ),
                  ),
                ),
              ],
          body: ListView.builder(itemBuilder: (context, index) {
            return Text('Sample Text: $index');
          })),
    );
  }
}
```
## Logs

```
```

`flutter analyze`:
```
Analyzing testing_app...                                         
No issues found! (ran in 1.2s)
```

`flutter doctor -v` (current beta // stable channel which doesn't have the issue):
```
[✓] Flutter (Channel beta, v1.0.0, on Mac OS X 10.14.2 18C54, locale de-DE)
    • Flutter version 1.0.0 at /Users/USERNAME/flutter
    • Framework revision 5391447fae (5 weeks ago), 2018-11-29 19:41:26 -0800
    • Engine revision 7375a0f414
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at /Users/USERNAME/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    • ios-deploy 1.9.4
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 31.3.1
    • Dart plugin version 181.5656
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
```

`flutter doctor -v` (current dev channel which has the issue):
```
[✓] Flutter (Channel dev, v1.1.4, on Mac OS X 10.14.2 18C54, locale de-DE)
    • Flutter version 1.1.4 at /Users/USERNAME/flutter
    • Framework revision 7a88fbc5fd (13 days ago), 2018-12-20 11:38:07 -0800
    • Engine revision dbdb5e6f8c
    • Dart version 2.2.0 (build 2.2.0-dev.1.1 f9ebf21297)
...
```

Screenshots (in fade / overlap):

Previously (title fading out when NestedScrollView is scrolled):
bildschirmfoto 2019-01-03 um 02 27 49
Video: https://youtu.be/ozBWfNvpIz8

Now (title not fading out when NestedScrollView is scrolled):
bildschirmfoto 2019-01-03 um 02 24 52
Video: https://youtu.be/YMG1SZmA8Bw

### PR DESCRIPTION
**Pull request by goderbauer**
Fixes https://github.com/flutter/flutter/issues/25993.
Follow-up to https://github.com/flutter/flutter/pull/25051.

### CODE REVIEW

**Comment by goderbauer** [comment]
@fkorotkov All Windows jobs are failing with:

```
Failed to start: FAILED_PRECONDITION: The image 'cirrusci/windowsservercore:2016' in container group 'cirrus-ci-task-5752589016629248' is not accessible. Please check the image and registry credential.
```

How can we fix that?

File: packages/flutter/lib/src/material/app_bar.dart [hunk 1]
```diff
@@ -650,8 +650,22 @@ class _SliverAppBarDelegate extends SliverPersistentHeaderDelegate {
...
+    // pinned | floating | bottom != null || opacity
+    // ----------------------------------------------
+    //    0   |    0     |        0       ||  fade
+    //    0   |    0     |        1       ||  fade
+    //    0   |    1     |        0       ||  fade
+    //    0   |    1     |        1       ||  fade
+    //    1   |    0     |        0       ||  1.0
+    //    1   |    0     |        1       ||  1.0
+    //    1   |    1     |        0       ||  1.0
+    //    1   |    1     |        1       ||  fade
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/material/app_bar.dart]: 💯

**Comment by fkorotkov** [comment]
@goderbauer seems Docker Hub is down https://status.docker.com/ 🤷‍♂️

**Review by GaryQian — APPROVED** [review]
LGTM!

File: packages/flutter/lib/src/material/app_bar.dart [hunk 2]
```diff
@@ -650,8 +650,22 @@ class _SliverAppBarDelegate extends SliverPersistentHeaderDelegate {
   @override
   Widget build(BuildContext context, double shrinkOffset, bool overlapsContent) {
     final double visibleMainHeight = maxExtent - shrinkOffset - topPadding;
-    final double toolbarOpacity = pinned ? 1.0
-      : ((visibleMainHeight - _bottomHeight) / kToolbarHeight).clamp(0.0, 1.0);
+
+    // Truth table for `toolbarOpacity`:
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/app_bar.dart]: It looks like the result is fade when pinned=false and 1.0 when pinned=true except with one exception (the last row). Aha: I see that the implementation boils it down nicely:
```
!pinned || (floating && bottom != null)
```

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**